photutils/detection/findstars.py

```diff
@@ -140,37 +106,126 @@ def daofind(data, threshold, fwhm, ratio=1.0, theta=0.0, sigma_radius=1.5,
     References
     ----------
 
-    .. [1] Stetson, P. 1987; PASP 99, 191 (http://adsabs.harvard.edu/abs/1987PASP...99..191S)
+    .. [1] Stetson, P. 1987; PASP 99, 191 (http://adsabs.harvard.edu/abs/
+           1987PASP...99..191S)
     .. [2] http://iraf.net/irafhelp.php?val=daofind&help=Help+Page
     .. [3] http://stsdas.stsci.edu/cgi-bin/gethelp.cgi?daofind
     """
 
-    daofind_kernel = _FindObjKernel(fwhm, ratio, theta, sigma_radius)
-    threshold *= daofind_kernel.relerr
-    objs = _findobjs(data, threshold, daofind_kernel,
-                     exclude_border=exclude_border)
-    tbl = _daofind_properties(objs, threshold, daofind_kernel, sky)
-    if len(objs) == 0:
-        warnings.warn('No sources were found.', AstropyUserWarning)
-        return tbl     # empty table
-    table_mask = ((tbl['sharpness'] > sharplo) &
-                  (tbl['sharpness'] < sharphi) &
-                  (tbl['roundness1'] > roundlo) &
-                  (tbl['roundness1'] < roundhi) &
-                  (tbl['roundness2'] > roundlo) &
-                  (tbl['roundness2'] < roundhi))
-    tbl = tbl[table_mask]
-    idcol = Column(name='id', data=np.arange(len(tbl)) + 1)
-    tbl.add_column(idcol, 0)
-    if len(tbl) == 0:
-        warnings.warn('Sources were found, but none pass the sharpness and '
-                      'roundness criteria.', AstropyUserWarning)
-    return tbl
-
+    def __init__(self, threshold, fwhm, ratio=1.0, theta=0.0,
+                 sigma_radius=1.5, sharplo=0.2, sharphi=1.0, roundlo=-1.0,
+                 roundhi=1.0, sky=0.0, exclude_border=False):
+        """
```

Comment:
We usually document the parameters for __init__ in the class docstring. That way, sphinx will render them properly. (Methods starting with one or two underscores ad usually not included in the sphinx output).